src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,174 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.exception.ParseException;
+import reposense.frontend.CliArguments;
+import reposense.util.TestUtil;
+
+public class ArgsParserTest {
+
+    private static final String INPUT_DELIMITER = " ";
+    private static final String PROJECT_DIRECTORY = System.getProperty("user.dir");
+
+    private static final String CONFIG_FILE_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "build").toString();
+    private static final String CONFIG_FILE_RELATIVE = Paths.get("sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_RELATIVE = Paths.get("build").toString();
+    private static final String DEFAULT_MANDATORY_ARGS = "-config " + CONFIG_FILE_ABSOLUTE + " ";
+
+    @Test
+    public void parse_allCorrectInputs_success() throws ParseException {
+        String input = DEFAULT_MANDATORY_ARGS + String.format("-output %s -since 01/07/2017 -until 30/11/2017",
+                OUTPUT_DIRECTORY_ABSOLUTE);
+        CliArguments cliArguments = ArgsParser.parse(input.split(INPUT_DELIMITER));
+        Assert.assertEquals(CONFIG_FILE_ABSOLUTE, cliArguments.getConfigFilePath().toString());
+        Assert.assertEquals(OUTPUT_DIRECTORY_ABSOLUTE, cliArguments.getTargetFilePath().toString());
+
+        Date expectedSinceDate = TestUtil.getDate(2017, Calendar.JULY, 1);
+        Date expectedUntilDate = TestUtil.getDate(2017, Calendar.NOVEMBER, 30);
+        Assert.assertEquals(expectedSinceDate, cliArguments.getSinceDate().get());
+        Assert.assertEquals(expectedUntilDate, cliArguments.getUntilDate().get());
+    }
+
+    @Test
+    public void parse_configFileOnly_success() throws ParseException {
+        String input = String.format("-config %s", CONFIG_FILE_ABSOLUTE);
+        CliArguments cliArgumentsOfAbsolutePath =  ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        input = String.format("-config %s", CONFIG_FILE_RELATIVE);
+        CliArguments cliArgumentsOfRelativePath = ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        Path expected = Paths.get(CONFIG_FILE_ABSOLUTE);
+        Assert.assertEquals(expected, cliArgumentsOfAbsolutePath.getConfigFilePath());
+        Assert.assertEquals(expected, cliArgumentsOfRelativePath.getConfigFilePath());
+
+        //Optional arguments have default values
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfAbsolutePath.getSinceDate());
+        Assert.assertEquals(Optional.empty(), cliArgumentsOfAbsolutePath.getUntilDate());
+        Assert.assertEquals(new File(".").getAbsolutePath(), cliArgumentsOfAbsolutePath.getTargetFilePath().toString());
```

Comment:
Use `new File(".").toPath()` to test using `Path`

Reply:
Actually, from this [SO](https://stackoverflow.com/questions/29368308/java-nio-how-is-files-issamefile-different-from-path-equals), `Paths.equals(Object)` may not return `true` even if both `Path` points to the same file/directory, instead we should use `Files.isSameFile(Path, Path)` here.